### PR TITLE
ST-144 - add Helm Chart based on K8s guidelines

### PR DESCRIPTION
**Pull request by piersharding**
ST-144 https://jira.skatelescope.org/browse/ST-144 - Refactor Tango-example according to the guideline for K8S 
* add Helm Chart
* add helper targets in Makefile
* add example gitlab-ci job steps

### CODE REVIEW

**Comment by piersharding** [comment]
I've added in a `helm test` example, and the machinery for a locally running Tiller instead of relying on a cluster install.  Helm is going to go v3 within a few months (at a guess)  as alpha has just been announced - https://github.com/helm/helm/releases .  This will change the setup requirements for Helm and might have a few minor changes on chart design.  Given this, I think the included examples will work for us over the bridging gap .

File: charts/tango-example/templates/tango-rest.yaml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      - name: tango-rest
+        image: "{{ .Values.tangorest.image.registry }}/{{ .Values.tangorest.image.image }}:{{ .Values.tangorest.image.tag }}"
+        imagePullPolicy: {{ .Values.tangorest.image.pullPolicy }}
+        command:
+          - /usr/local/bin/wait-for-it.sh
+          - databaseds-{{ template "tango-example.name" . }}-{{ .Release.Name }}:10000
+          - --timeout=30
+          - --strict
+          - --
+          - /usr/bin/supervisord
```
Comment (flyingfrog81) [inline charts/tango-example/templates/tango-rest.yaml]: If supervisord is used, how are responsibilities different from K8s orchestration? Shall we settle on just one of the two for managing services as OS processes or running containers?

**Comment by Matteo04052017** [comment]
Not sure why we are keeping the tango-rest service. The Button team decided to go for the webjive and not for this... is there any reason for keeping it?

**Comment by flyingfrog81** [comment]
LGTM, added a couple of questions mostly for me to better understand.

**Comment by flyingfrog81** [comment]
> Not sure why we are keeping the tango-rest service. The Button team decided to go for the webjive and not for this... is there any reason for keeping it?

none that I know of .... can you remove it at least from this k8s branch?

**Comment by Matteo04052017** [comment]
Since I don't know why it has been kept, I would prefer to wait for Piers to do it.

File: charts/tango-example/templates/tango-rest.yaml (hunk 1 above)
Comment (piersharding) [reply charts/tango-example/templates/tango-rest.yaml]: Good point - I forgot about supervisord being there and IMO we should not be doing this kind of thing - pushed up a change for this.

**Comment by piersharding** [comment]
> > Not sure why we are keeping the tango-rest service. The Button team decided to go for the webjive and not for this... is there any reason for keeping it?
> 
> none that I know of .... can you remove it at least from this k8s branch?

I guess this is an application specific question as to whether it is of any value to keep.  From my point of view it provided a useful (and stateless) interface to test against using `curl` in the `make ingress_check` target (makes it a good candidate for automated tests) without having to launch the heavy weight webjive alternative.

Happy to remove it if it is not required?

**Comment by piersharding** [comment]
> Trying to undesrstand how scalable is this approach for integrating a new device. The `powersupply.yaml` file seems to be all boilerplate code, is anything device specific going into this file regarding its configuration?
> Would it be possible (not now) to split the `values.yaml` into a devices file and a generic k8s configuration values file that the developer should not care about?

This is where Helm charts excel - the devices can be described in a values structure like:
```
devices:
  - name: pwrsp1
    instance: PowerSupply/test
    class: PowerSupply
    devlist: test/power_supply/1
  - name: pwrsp2
    instance: PowerSupply/test
    class: PowerSupply
    devlist: test/power_supply/2
  - name: pwrspN
    instance: PowerSupply/test
    class: PowerSupply
    devlist: test/power_supply/N
...
```
And then the whole template of tango-example/templates/powersupply.yaml can be put in a helper or a partial (https://helm.sh/docs/chart_template_guide/#partials-and-files) and the data structure iterated over with a `Range` - https://helm.sh/docs/chart_template_guide/#looping-with-the-range-action.

**Comment by Matteo04052017** [comment]
Ok, if you want use it, then I have no problem to keep it.

**Comment by flyingfrog81** [comment]
I'm happy with the changes and the answers to all my questions, thanks! 
Mergin this.

**Pull request merged**